src/shared/conf-parser.c

```diff
@@ -543,13 +542,11 @@ static int config_parse_many_files(
         STRV_FOREACH(fn, conf_files) {
                 _cleanup_fclose_ FILE *f = NULL;
 
-                f = fopen(*fn, "re");
-                if (!f) {
-                        if (errno == ENOENT)
-                                continue;
-
-                        return -errno;
-                }
+                r = chase_and_fopen_unlocked(*fn, root, CHASE_AT_RESOLVE_IN_ROOT, "re", NULL, &f);
```

Comment:
Also here: we need final path and use it instead of `*fn`.

Reply:
I don't think so. It's OK if we chase `*fn` again. We're not trying to prevent a race condition, we just want to read the files safely. If things are being modified underneath while we're reading the config, the only thing that is required is that we don't go out above `root`.